onchain/src/art_peace.cairo

```diff
@@ -962,6 +972,31 @@ pub mod ArtPeace {
                             // TODO: Check if the color is transparent
                             if color == self.canvas.read(pos).color {
                                 matches += 1;
+
+                                let mut pixel_owner = self.canvas.read(pos).owner;
+                                let mut found = false;
+                                let mut user_index = 0;
+                                let mut index = 0;
+
+                                while index < pixel_contributors_span
```

Comment:
pixel_contributors could get pretty large, so searching through it each time would probably be inefficient.

Instead, it might be better to use 2 maps and an array :  
`pixel_contributors_indexes: Felt252Dict<u32>`, `total_pixels_by_user: Felt252Dict<u32>`, and `pixel_contributors: Array<ContractAddress>` where :
`pixel_contributors_indexes` maps user address -> idx in t`otal_pixels_by_user` map & `pixel_contributors` array
`total_pixels_by_user` maps idx starting at 1 -> total pixels

Then instead of searching for the user in `pixel_contributors` you can just check if `pixel_contributors_indexes.at(user address)` == 0, and if so, then add the users info to total pixels and pixel contributors maps and arrays.